airbyte_cdk/sources/declarative/models/declarative_component_schema.py

```diff
@@ -1831,6 +1831,21 @@ class DynamicSchemaLoader(BaseModel):
         description="Component used to coordinate how records are extracted across stream slices and request pages.",
         title="Retriever",
     )
+    schema_transformations: Optional[
```

Comment:
nit: isn't it redundant to name it schema_transformations, as we are already in the schema area? Other fields are not named schema_title or schema_retriever

Reply:
This component is used to transform a schema. If we refer to it simply as a 'transformation', it could be misleading. It might suggest that transformations are used to alter retrieved records, which is not the case. The actual process involves first retrieving the record and, if necessary (such as when a schema API returns more than one schema), filtering them. Only after extracting the schema from the retrieved records do we proceed to transform it.

Reply:
I'm ok leaving it as schema_transformations, but not sure it would be misleading as the DynamicSchemaLoader context would provide sufficient context IMO but not a reason to hold this pr for this.